config/schemas/config.yaml

```diff
@@ -0,0 +1,2719 @@
+    items:
+      type: string
+      oneOf:
+        - format: ipv4
+        - format: ipv6
+        - format: netmask
+    type: array
+  cpumem:
+    title: CPU and Memory specifications
+    properties:
+      cpu:
+        default: 100m
+        pattern: ^[1-9][0-9]*m?$
+        type:
+          - string
+          - integer
+      memory:
+        default: 128Mi
+        pattern: ^[0-9]+(\.[0-9]+)?([KMG]i)?$
+        type:
+          - string
+          - integer
+    type: object
+  listen:
+    properties:
+      ips:
+        $ref: '#/$defs/iplist'
+      ports:
+        $ref: '#/$defs/portlist'
+    type: object
+  nodeSelector:
+    default:
+      kubernetes.io/os: linux
+    properties:
+      kubernetes.io/os:
+        default: linux
+        type: string
+    type: object
+  portlist:
+    items:
+      maximum: 65535
+      minimum: 1
+      type: integer
+    type: array
+  resources:
+    properties:
+      limits:
+        $ref: '#/$defs/cpumem'
+      requests:
+        $ref: '#/$defs/cpumem'
+    type: object
+  tolerations:
+    items:
+      properties:
+        effect:
+          default: NoSchedule
+          type: string
+        key:
+          default: node-role.kubernetes.io/master
+          type: string
+        operator:
+          default: Exists
+          type: string
+      type: object
+    type: array
+  topologySpreadConstraints:
+    items:
+      properties:
+        labelSelector:
+          properties:
+            matchLabels:
+              additionalProperties:
+                type: string
+              type: object
+          type: object
+        maxSkew:
+          default: 1
+          type: number
+        topologyKey:
+          default: kubernetes.io/hostname
+          type: string
+        whenUnsatisfiable:
+          default: DoNotSchedule
+          type: string
+      type: object
+    type: array
+  commonResource:
+    $comment: |
+      Uncertain whether it's possible to combine $ref and additional
+      properties, short of using 'allOf'
+    type: object
+    properties:
+      affinity:
+        $ref: '#/$defs/affinity'
+      nodeSelector:
+        $ref: '#/$defs/nodeSelector'
+      resources:
+        $ref: '#/$defs/resources'
+      tolerations:
+        $ref: '#/$defs/tolerations'
+  clusterid:
+    type: string
+    enum:
+      - sc
+      - wc
+  timeRange:
+    title: An amount of time
+    type: string
+    pattern: ^[0-9]+[hmsd]+$
+    examples:
+      - 300s
+      - 72h
+      - 3dd
+  goDuration:
+    title: A more complexly expressed amount of time
+    type: string
+    pattern: ^([-+]?[0-9]+(.[0-9]+)?(ns|us|µs|ms|s|m|h))+$
+    examples:
+      - 2h45m0s
+type: object
+required:
+  - global
+properties:
+  alerts:
+    properties:
+      alertTo:
+        type: string
+      opsGenie:
+        properties:
+          apiUrl:
+            # These 'default' values are mostly lifted from existing config files.
+            default: https://api.eu.opsgenie.com
```

Comment:
Will the validator add this value if apiUrl is empty? Or how does default work? :smile:

Reply:
Editor integration could use defaults for autocomplete, for example [this one](https://github.com/redhat-developer/vscode-yaml) claims to do so.